### PR TITLE
Simple Anagram Helper

### PR DESCRIPTION
**Pull request by oliverabrahams**
## What are you changing?

- Simplify Anagram helper to a similar experience to what we have at the moment in prod
- Will build on this but for now this will be sufficient and we can move forward to shipping

Screenshot 2024-12-09 at 15 28 52
Screenshot 2024-12-09 at 15 29 03
Screenshot 2024-12-09 at 15 29 24
Screenshot 2024-12-09 at 15 29 38

### CODE REVIEW

File: libs/@guardian/react-crossword/src/utils/biasedShuffle.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { isUndefined } from '@guardian/libs';
+
+// This uses the Fisher Yates shuffle algorithm, but with a bias factor.
```
Comment (sndrs) [inline libs/@guardian/react-crossword/src/utils/biasedShuffle.ts:3]: 🥰

File: libs/@guardian/react-crossword/src/components/AnagramHelper.stories.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import type { Meta, StoryObj } from '@storybook/react';
 import { groupedClues as data } from '../../stories/formats/grouped-clues';
+import { progressTwo } from '../../stories/formats/grouped-clues.progress';
```
Comment (sndrs) [inline libs/@guardian/react-crossword/src/components/AnagramHelper.stories.tsx:3]: 🤪

Comment (oliverabrahams) [reply libs/@guardian/react-crossword/src/components/AnagramHelper.stories.tsx]: oopsy

File: libs/@guardian/react-crossword/src/components/AnagramHelper.tsx [hunk 3]
```diff
@@ -1,112 +1,71 @@
...
-	const [shuffled, setShuffled] = useState<boolean>(false);
-	const [candidateLetters, setCandidateLetters] = useState<string[]>([]);
-	const [wordWheelLetters, setWordWheelLetters] = useState<string[]>([]);
-	const [progressLetters, setProgressLetters] = useState<
-		AnagramHelperProgress[]
-	>([]);
-	const { entries } = useData();
-	const { progress } = useProgress();
-	const { updateCell } = useUpdateCell();
+	const [letters, setLetters] = useState<string>('');
```
Comment (sndrs) [inline libs/@guardian/react-crossword/src/components/AnagramHelper.tsx:18]: if you set a default value, i don't think you need to type it? not a blocker though

File: libs/@guardian/react-crossword/src/components/AnagramHelper.tsx [hunk 4]
```diff
@@ -132,56 +91,79 @@ export const AnagramHelper = () => {
...
+						</span>
+					</div>
+				)}
+				{solving && (
+					<>
+						<WordWheel letters={shuffledLetters} />
+						<div
+							css={css`
+								margin: ${space[4]}px ${space[1]}px;
+								* {
```
[the remark this answers is not in the document]
Comment (oliverabrahams) [reply libs/@guardian/react-crossword/src/components/AnagramHelper.tsx:139]: I think so. Do you think i should make it button specific?

**Review by sndrs — APPROVED** [review]
nice

File: libs/@guardian/react-crossword/src/components/AnagramHelper.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (sndrs) [reply libs/@guardian/react-crossword/src/components/AnagramHelper.tsx:143]: yeah i think could catch someone in the future (e.g. a button with an icon in it?)

**Pull request merged**